dali/base/dasds.cpp

```diff
@@ -3675,6 +3790,78 @@ class CLockInfo : public CInterface, implements IInterface
         return out;
     }
 
+    bool getFilteredLocks(CLockFilters &filters, MemoryBuffer &mb)
+    {
+        bool isFile;
+        StringBuffer file;
+        if (!filters.checkAndReadPath(xpath, isFile, file))
+            return false;
+
+        unsigned nlocks=0;
+        MemoryBuffer locks, localMB;
+        UInt64Array keys;
+        {
+            CHECKEDCRITICALBLOCK(crit, fakeCritTimeout);
+            HashIterator iter(connectionInfo);
+            ForEach(iter)
+            {
+                IMapping &imap = iter.query();
+                LockData *lD = connectionInfo.mapToValue(&imap);
+                keys.append(* ((ConnectionId *) imap.getKey()));
+                locks.append(sizeof(LockData), lD);
+                ++nlocks;
+            }
+        }
+
+        if (nlocks < 1)
+            return false;
+
+        if (isFile)
```

Comment:
I think this is embedded too low down, this method should serialize the info out (filtered) and then it should be interpreted outside.

I think you should just add xpath here.

Actually I think the serialization of this lock info in general needs reworking, I'll add a comment in PR in a moment.